docs/memory.md

```diff
@@ -439,7 +439,9 @@ workload run more efficiently.
 
 Multiple values can be provided to define the list. Each value is a string
 referring to an existing memory zone identifier. Values are separated from
-each other with the `:` separator.
+each other with the `:` separator. Note that the memory zone id used by
+different NUMA nodes should also be different, i.e. this is not allowed:
+`--numa guest_numa_id=0,memory_zones=mem0 guest_numa_id=1,memory_zones=mem0`
```

Comment:
Hi @sboeuf, could you please confirm this change is correct? As from my understanding, different NUMA id cannot share the same memory zone...If this is correct, should we also add some code to forbid user setting command line like this? Thanks!

Reply:
Yes you're totally right. We can't have the same memory zone belonging to more than one NUMA node. If this is not validated in the code, then yes we should add a check to prevent the user from specifying such configuration.

Reply:
I'd rephrase with:
```
Note that a memory zone must belong to a single NUMA node. The following
configuration is incorrect, therefore not allowed:
`--numa guest_numa_id=0,memory_zones=mem0 guest_numa_id=1,memory_zones=mem0`
```

Reply:
> Yes you're totally right. We can't have the same memory zone belonging to more than one NUMA node. If this is not validated in the code, then yes we should add a check to prevent the user from specifying such configuration.

AFAIK the current code is not validating that - As I remember I tried this once...

Then I will add some checking code. How about this? @sboeuf

Here (https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/vmm/src/config.rs#L1689) when we parsing the config, we add a global/static data structure, for example a vector, to store the already used memory zones. If we meet a used memory zone when parsing the rest of NUMA nodes, we raise an error.